golioth/golioth.py

```diff
@@ -776,12 +787,14 @@ async def delete(self, artifact_id: str):
     async def upload(self,
                      path: Path,
                      version: str,
-                     package: str = 'main') -> Artifact:
+                     package: str = 'main',
+                     blueprint_id: str = None) -> Artifact:
```

Comment:
If `blueprint_id` is `str`, then it cannot be `None`. You should use `str | None` or `Optional[str]` instead.